ui/src/app/views/account-creation/component.ts

```diff
@@ -101,6 +93,34 @@ export class AccountCreationComponent {
     });
   }
 
+  get showPasswordsDoNotMatchError() {
+    // We do not want to show errors if nothing is typed yet. This is caught by the required
+    // fields case.
+    if (this.password === undefined || this.passwordAgain === undefined
+      || this.password.length === 0 || this.passwordAgain.length === 0) {
+        return false;
+    }
+    return !(this.password === this.passwordAgain);
+  }
+
+  get showPasswordLengthError() {
+    // We do not want to show errors if nothing is typed yet. This is caught by the required
+    // fields case.
+    if (this.password === undefined || this.password.length === 0) {
+      return false;
+    }
+    return (this.password.length < 8 || this.password.length > 100);
+  }
+
+  get containsLowerAndUpperError() {
+    // We do not want to show errors if nothing is typed yet. This is caught by the required
+    // fields case.
+    if (this.password === undefined || this.password.length === 0) {
+      return false;
+    }
+    return !(this.hasLowerCase(this.password) && this.hasUpperCase(this.password));
+  }
+
   get usernameInvalidError(): boolean {
```

Comment:
I'm a bit worried about conflating validity of the field, and whether we are currently displaying the error. Not sure how this could be improved, perhaps with a naming convention, e.g. `show*` vs `is*`

Reply:
Yeah. May not need the latter one if there's not a need. Concretely maybe renaming to `showContainsLowerAndUpperError` and `showUsernameInvalidError`

The point I was getting at is that there's conceptually two things here:
a. "is this field valid"?
b. "should I currently display an error for this field/constraint?"

The concern is that some utilities are calling helpers which answer (b), e.g. `usernameValidationSuccess` and the `createAccount` validation logic. So you could consider an approach more like:

```
isUsernameInvalid(): boolean {
     // Include alphanumeric characters, -'s, _'s, apostrophes, and single .'s in a row.	     // Include alphanumeric characters, -'s, _'s, apostrophes, and single .'s in a row.
     return !(new RegExp(/^[\w-']([.]{0,1}[\w-']+)*$/).test(username));	     return !(new RegExp(/^[\w-']([.]{0,1}[\w-']+)*$/).test(username));
}

showUsernameInvalid(): boolean {
   if (isBlank(username)) {
     return false;
   }
  return this.isUsernameInvalid();
}

// Other utils call isUsernameInvalid
```

I'm not sure it makes sense for all of the validation rules here, but food for thought.